lib/point_quest_web/live/quest.ex

```diff
@@ -19,7 +31,53 @@ defmodule PointQuestWeb.QuestLive do
   end
 
   def mount(params, _session, socket) do
+    classes = PointQuest.Quests.Adventurer.Class.NameEnum.valid_atoms()
+
+    changeset = get_changeset(%{quest_id: params["id"]})
+
     {:ok, quest} = Infra.Quests.Db.get_quest_by_id(params["id"])
-    {:ok, assign(socket, session_id: params["id"], quest: quest)}
+
+    {:ok,
+     assign(socket,
+       session_id: params["id"],
+       quest: quest,
+       form: to_form(changeset),
+       classes: classes
+     )}
+  end
+
+  def handle_event("validate_adventurer", %{"add_adventurer" => params}, socket) do
+    params = Map.put(params, "quest_id", socket.assigns.quest.id)
+
+    form =
+      get_changeset(params)
+      |> to_form()
+
+    {:noreply, assign(socket, form: form)}
+  end
+
+  def handle_event(
+        "join_party",
+        %{"add_adventurer" => %{"name" => name, "class" => class}},
+        socket
+      ) do
+    {:ok, quest} =
+      AddAdventurer.new!(%{quest_id: socket.assigns.quest.id, name: name, class: class})
+      |> AddAdventurer.execute()
+
+    adventurer = Enum.find(quest.adventurers, fn a -> a.name == name end)
+
+    token =
+      PointQuest.Authentication.create_actor(adventurer)
```

Comment:
**Nit (non-blocking)-** Embrace the PIPE!

With these longer context names I am feeling like the single pipe looks a bit untidy.
What do you think about
```elixir
token = adventurer
|> PointQuest.Authentication.create_actor()
|> PointQuest.Authentication.actor_to_token()
```

I have been feeling this way about the command calls too.

Reply:
Totes fair. Will do a quick refactorino and get this pushed up.